src/query/formats/src/file_format_type.rs

```diff
@@ -324,6 +325,14 @@ fn check_options(options: &mut FileFormatOptions) -> Result<()> {
         options.record_delimiter = '\n'.to_string();
     }
 
+    if options.nan_display.is_empty() {
+        options.nan_display = "NaN".to_string();
+    } else if options.nan_display.len() != 3 || options.nan_display.to_lowercase() != "nan" {
+        return Err(ErrorCode::InvalidArgument(
+            "nan_display must be case-sensitive literal `nan`",
```

Comment:
case-insensitive?

Reply:
case-sensitive is right here, eg: nan, NaN, Nan.